### PR TITLE
Groups & Assignments API implementation with view & viewmodel bindings

### PR DESCRIPTION
**Pull request by Nitish145**

### CODE REVIEW

**Comment by Nitish145** [comment]
@tachyons @nitin10s please review

**Review by nitin10s — CHANGES_REQUESTED** [review]
1.The Edit group button currently is just a pencil icon. Make it similar to the current platform [Text: Edit along with the icon]. Same goes for edit assignment section.
2. Make a new group button has a different style than all other buttons. Please rectify. 
3. When I try to make an assignment with some elements restricted, the screen gets stuck on the loading state saying adding.
4. Edit Group Name section: The button SAVE is written in all caps, Please use sentence case.

File: lib/models/assignments.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  String id;
+  String type;
+  AssignmentAttributes attributes;
+  List<Project> projects;
+  List<Grade> grades;
+
+  factory Assignment.fromJson(Map<String, dynamic> json) => Assignment(
+        id: json['id'] ?? json['data']['id'],
+        type: json['type'] ?? json['data']['type'],
+        attributes: AssignmentAttributes.fromJson(
```
Comment (tachyons) [inline lib/models/assignments.dart:40]: https://pub.dev/packages/json_api use libraries for serialization

Comment (Nitish145) [reply lib/models/assignments.dart:40]: Will see to it, a bit of research would be required to get to a conclusion if this could a good fit.. @tachyons

**Review by nitin10s — APPROVED** [review]

**Comment by Nitish145** [comment]
@nitin10s the assignments bug is fixed, please check!!

**Comment by Nitish145** [comment]
@tachyons please check!!

**Pull request merged**